datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparison.scala

```diff
@@ -166,22 +166,13 @@ class DatasetComparison(cliOptions: CliOptions,
   }
 
   /**
-   * Handles duplicates in a sense that this method looks for them. Then based on the configuration if found, throws an
-   * error, does deduplication or just passes through.
+   * Handles duplicates in a sense that this method looks for them based on a primary key. Then based on the
+   * configuration, if found, throws an error.
    *
    * @param dfsWithKey DataFrame pair where both have appended unique key
-   * @return
+   * @return A Pair for comparison counts
    */
   private def handleDuplicates(dfsWithKey: ComparisonPair[DataFrame], cmpUniqueColumn: String): ComparisonPair[Long] = {
```

Comment:
I would wonder why input both dataframes in comparison as a pair. Has no functional reason, just can confuse...

Reply:
This was decided before the 0.2.0 release. That since we handle everything in a pair, we created a `ComparisonPair` case class to move stuff around. The clutter created from moving around multiple alone pairs and was much worse.

`ComparisonPair` on the other hand forces you to always use the correct pair of data. Be it Long, DataFrame or others. It actually made coding for me quite easier.

Reply:
OK, fair enough